pkg/tls/config.go

```diff
@@ -45,54 +42,72 @@ var (
+	version, err := validatedMinVersion(minVersion)
 	if err != nil {
-		return nil, fmt.Errorf("error while configuring minimal TLS version: %s", err)
+		return nil, err
 	}
-	// Get configured tls ciphers
-	TLSCiphers, err := lookupTLSCiphers()
+	cipherSuites, err := validatedCiphers(ciphers, version)
 	if err != nil {
-		return nil, fmt.Errorf("error while configuring TLS ciphers: %s", err)
+		return nil, err
 	}
-
 	return &tls.Config{
-		PreferServerCipherSuites: true,
-		MinVersion:               TLSMinVersion,
-		CipherSuites:             TLSCiphers,
+		MinVersion:   version,
+		CipherSuites: cipherSuites,
 	}, nil
 }
 
-func lookupTLSVersion() (uint16, error) {
-	tlsVersionKeys := getKeysFromMap(tlsVersions)
-	settingsTLSMinVersion := settings.TLSMinVersion.Get()
-	if val, ok := tlsVersions[settingsTLSMinVersion]; ok {
+func validatedMinVersion(version string) (uint16, error) {
+	if val, ok := validVersions[strings.TrimSpace(version)]; ok {
 		return val, nil
 	}
-	return 0, fmt.Errorf("invalid minimal TLS version [%s], must be one of: %s", settingsTLSMinVersion, strings.Join(tlsVersionKeys, " "))
+	valid := keysFromMap(validVersions)
+	sort.Strings(valid)
+	return 0, fmt.Errorf("unsupported minimal TLS version %s, must be one of: %s", version, strings.Join(valid, ", "))
 }
 
-func lookupTLSCiphers() ([]uint16, error) {
-	tlsCipherSuitesKeys := getKeysFromMap(tlsCipherSuites)
-	settingsTLSCiphers := settings.TLSCiphers.Get()
-	sliceTLSCiphers := strings.Split(settingsTLSCiphers, ",")
+func validatedCiphers(ciphers string, version uint16) ([]uint16, error) {
+	split := strings.Split(ciphers, ",")
+	if version == tls.VersionTLS13 {
+		return validatedCiphersHelper(split, validCiphers13)
+	}
+	return validatedCiphersHelper(split, union(validCiphers10to12, validCiphers13))
+}
 
-	var TLSCiphers []uint16
-	for _, TLSCipher := range sliceTLSCiphers {
-		val, ok := tlsCipherSuites[strings.TrimSpace(TLSCipher)]
+func validatedCiphersHelper(ciphers []string, validCiphers map[string]uint16) ([]uint16, error) {
+	tlsCiphers := make([]uint16, 0, len(ciphers))
+	for _, cipher := range ciphers {
+		c, ok := validCiphers[strings.TrimSpace(cipher)]
 		if !ok {
-			return []uint16{}, fmt.Errorf("unsupported cipher [%s], must be one or more from: %s", TLSCipher, strings.Join(tlsCipherSuitesKeys, " "))
+			valid := keysFromMap(validCiphers)
+			sort.Strings(valid)
+			return nil, fmt.Errorf("unsupported cipher %s, must be one or more of:\n%s", cipher, strings.Join(valid, "\n"))
+		}
+		tlsCiphers = append(tlsCiphers, c)
+	}
+	return tlsCiphers, nil
+}
+
+func union(cipherSets ...map[string]uint16) map[string]uint16 {
+	result := make(map[string]uint16)
+	for _, set := range cipherSets {
+		for k, v := range set {
+			result[k] = v
 		}
-		TLSCiphers = append(TLSCiphers, val)
 	}
-	return TLSCiphers, nil
+	return result
 }
 
-func getKeysFromMap(input map[string]uint16) []string {
-	var keys []string
+func keysFromMap(input map[string]uint16) []string {
```

Comment:
Because it's a refactoring of this part of the package, I thought it's OK to change unexported functions to give them better names.

Reply:
It's not about exported vs unexported. My point is it's very subjective about what is better. IMO the old name was fine. If it doesn't enhance the functionality and isn't objectively more readable, it would be better to leave it alone. The more changes that are made, the harder it is to review and the greater the likelihood of introducing a bug.

Reply:
I definitely agree. I just think that in this case, it's permissible, since I planned on re-writing a relatively small piece of code, having added tests in advance as a safety net.

Reply:
As for good vs. bad names, I am not that obsessed with them, just thought its change is OK, given the opportunity to refactor.

Here is one example of a decent argument for verbs vs. nouns in method names: https://www.yegor256.com/2018/08/22/builders-and-manipulators.html

Reply:
@maxsokolovsky I understand your point and think it is a grey area on when to make stylistic changes, such as renaming variables or functions. If the change does not help maintainability, then it is not needed. This stops our history from being bloated with unnecessary name changes. And in this particular case, I don't think using a noun over a verb helps maintainability.